framework/aws-kms/aws-kms-hierarchical-keyring.md

```diff
@@ -294,66 +334,204 @@ in order to compute the [cache entry identifier](../cryptographic-materials-cach
+)
 ```
 
 ### Decryption Materials
 
 When the hierarchical keyring receives an OnDecrypt request,
-it MUST calculate the cache entry identifier as the first 32 bytes of the SHA-512 hash of the following byte strings, in the order listed:
+it MUST calculate the cache entry identifier as the SHA-384 hash of the following byte strings, in the order listed:
+
+- MUST be the Resource ID for the Hierarchical Keyring (0x02)
+- MUST be the Scope ID for Decrypt (0x02)
+- MUST be the Partition ID for the Hierarchical Keyring
+- Resource Suffix
+  - MUST be the UTF8 encoded Logical Key Store Name of the keystore for the Hierarchical Keyring
+  - MUST be the UTF8 encoded branch-key-id
+  - MUST be the UTF8 encoded branch-key-version
+
+All the above fields must be separated by a single NULL_BYTE `0x00`.
+
+| Field                  | Length (bytes) | Interpreted as      |
+| ---------------------- | -------------- | ------------------- |
+| Resource ID            | 1              | bytes               |
+| Null Byte              | 1              | `0x00`              |
+| Scope ID               | 1              | bytes               |
+| Null Byte              | 1              | `0x00`              |
+| Partition ID           | Variable       | bytes               |
+| Null Byte              | 1              | `0x00`              |
+| Logical Key Store Name | Variable       | UTF-8 Encoded Bytes |
+| Null Byte              | 1              | `0x00`              |
+| Branch Key ID          | Variable       | UTF-8 Encoded Bytes |
+| Null Byte              | 1              | `0x00`              |
+| branch-key-version     | 36             | UTF-8 Encoded Bytes |
 
-- MUST be the length of the branch key id in its uint8 interpretation
-- MUST be the UTF8 encoded branch-key-id
-- MUST be a single null byte `0x00`
-- MUST be the UTF8 encoded branch-key-version
-
-| Field                   | Length (bytes) | Interpreted as |
-| ----------------------- | -------------- | -------------- |
-| Length of branch-key-id | 3              | UInt8          |
-| branch-key-id           | Variable       | UTF-8 Encoded  |
-| Null Byte               | 1              | `0x00`         |
-| branch-key-version      | 36             | UTF-8 Encoded  |
+As a formula:
 
 ```
-branch-key-id = UTF8Encode(edk.providerInfo)
-
-ENTRY_ID = SHA512(
-    LengthUint8(branch-key-id) +
-    branch-key-id +
-    0x00 +
-    UTF8Encode(branch-key-version)
-)[0:32]
+resource-id = [0x02]
+scope-id = [0x02]
+logical-key-store-name = UTF8Encode(keystore.LogicalKeyStoreName)
+branch-key-id = UTF8Encode(hierarchicalKeyring.BranchKeyIdentifier)
+branch-key-version = UTF8Encode(branchKeyVersion)
+NULL_BYTE = [0x00]
+
+ENTRY_ID = SHA384(
+    resource-id
+    + NULL_BYTE
+    + scope-id
+    + NULL_BYTE
+    + partition-id
+    + NULL_BYTE
+    + logical-key-store-name
+    + NULL_BYTE
+    + branch-key-id
+    + NULL_BYTE
+    + branch-key-version
+)
 ```
 
+## Shared Cache Considerations
```

Comment:
Let's also add here the detail about the shared access to materials

Reply:
I added a brief note mentioning this here. Thanks!